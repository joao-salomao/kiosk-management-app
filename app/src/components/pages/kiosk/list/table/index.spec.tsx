import renderer from 'react-test-renderer';
import { Table, TableProps } from ".";

it('renders the kiosk list correctly', () => {
  const kiosks: TableProps['kiosks'] = [
    {
      id: "1",
      serialKey: "NDK-1",
      description: "Burguer King",
      isKioskClosed: false,
      storeOpensAt: "09:00:00",
      storeClosesAt: "22:00:00"
    },
    {
      id: "2",
      serialKey: "NDK-2",
      description: "Mac",
      isKioskClosed: false,
      storeOpensAt: "09:00:00",
      storeClosesAt: "22:00:00"
    }
  ]

  const tree = renderer
    .create(
      <Table kiosks={kiosks} isLoading={false} />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});


it('renders loader correctly', () => {

  const tree = renderer
    .create(
      <Table kiosks={[]} isLoading={true} />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});