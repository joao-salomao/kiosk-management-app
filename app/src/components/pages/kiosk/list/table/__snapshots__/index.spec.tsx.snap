// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loader correctly 1`] = `
<div
  className="overflow-x-auto shadow-md sm:rounded-lg"
>
  <table
    className="w-full text-sm text-left text-gray-500 text-gray-400"
  >
    <thead
      className="text-xs text-white uppercase bg-gray-50 bg-gray-700 text-gray-400"
    >
      <tr>
        <th
          className="px-6 py-3"
          scope="col"
        >
          ID
        </th>
        <th
          className="px-6 py-3"
          scope="col"
        >
          Description
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="bg-white border-b bg-gray-800 border-gray-700"
      >
        <th
          className="px-6 py-4 text-center"
          colSpan={2}
          scope="row"
        >
          Carregando
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renders the kiosk list correctly 1`] = `
<div
  className="overflow-x-auto shadow-md sm:rounded-lg"
>
  <table
    className="w-full text-sm text-left text-gray-500 text-gray-400"
  >
    <thead
      className="text-xs text-white uppercase bg-gray-50 bg-gray-700 text-gray-400"
    >
      <tr>
        <th
          className="px-6 py-3"
          scope="col"
        >
          ID
        </th>
        <th
          className="px-6 py-3"
          scope="col"
        >
          Description
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="bg-white border-b bg-gray-800 border-gray-700"
      >
        <th
          className="px-6 py-4"
          scope="row"
        >
          1
        </th>
        <td
          className="px-6 py-4"
        >
          Burguer King
        </td>
      </tr>
      <tr
        className="bg-white border-b bg-gray-800 border-gray-700"
      >
        <th
          className="px-6 py-4"
          scope="row"
        >
          2
        </th>
        <td
          className="px-6 py-4"
        >
          Mac
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
